frontend/packages/component-library/src/image/Image.tsx

```diff
@@ -21,6 +21,8 @@ export interface ImageProps
   onLoad?: () => void;
   /** Image onError callback */
   onError?: () => void;
+  /** Custom image component to replace the default <img> element */
+  imageComponent?: JSX.ElementType;
```

Comment:
For consistency with codebase I think it would be better to follow CamelCase naming for 'component' type props

```suggestion
  ImageComponent?: JSX.ElementType;
```